### PR TITLE
Rewrite xtdb-cli.py with "click"

### LINKED ISSUE #2866: Rewrite xtdb-cli.py in "click" and add documentation
Rewrite: https://github.com/minvws/nl-kat-coordination/blob/220670ebb898b3ab45659874ec75844ac70163d3/octopoes/tools/xtdb-cli.py#L1-L210 with "click" as demoed by @ammar92: https://github.com/minvws/nl-kat-coordination/blob/f82e97ebf7f7551513fc06f028a0eeaed14fe20b/octopoes/tools/xtdb-cli.py#L1-L263

And add documentation as per #2877.

### PR DESCRIPTION
**Pull request by originalsouth**
### Changes

Convert xtdb-cli.py to click (and add more features).

### Issue link

Closes #2866

### Demo

`octopoes/tools/xtdb-cli.py --help`

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication

- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by Donnype** [comment]
@originalsouth Could you update the PR title? :grin:

File: octopoes/octopoes/api/router.py [hunk 1]
```diff
@@ -412,10 +412,15 @@ def get_scan_profile_inheritance(
 ) -> list[InheritanceSection]:
     ooi = octopoes.get_ooi(reference, valid_time)
     if not ooi.scan_profile:
-        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="OOI does not have a scanprofile")
+        raise HTTPException(
```
Comment (Donnype) [inline octopoes/octopoes/api/router.py:415]: Are these format-only changes? Then these would probably be reverted when https://github.com/minvws/nl-kat-coordination/pull/2975 is merged?

File: octopoes/tools/xtdb_client.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
+import datetime
+
+import httpx
+from pydantic import JsonValue
+
+
+class XTDBClient:
```
Comment (Donnype) [inline octopoes/tools/xtdb_client.py:7]: Can't we reuse the client we have in `octopoes/xtdb`? I think that would make sense at least, even if it triggers some refactoring on the module side. Perhaps there's some inspiration to be found in https://github.com/Donnype/xtdb-py/blob/main/xtdb/session.py as well, but perhaps we are tied to some Octopoes specific stuff.

File: octopoes/tools/xtdb-cli.py [hunk 3]
```diff
@@ -1,219 +1,254 @@
...
+@click.pass_context
+def cli(ctx: click.Context, url: str, node: str, timeout: int, verbosity: int):
+    if verbosity == 1:
+        logging.basicConfig(level=logging.WARN)
+    elif verbosity == 2:
+        logging.basicConfig(level=logging.INFO)
+    elif verbosity == 3:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        raise click.UsageError("Invalid verbosity level (use -v, -vv, or -vvv)")
```
Comment (underdarknl) [inline octopoes/tools/xtdb-cli.py:44]:
```suggestion
    verbosities = [logging.WARN, logging.INFO, logging.DEBUG]
    try:
        logging.basicConfig(level=verbosities[verbosity-1])
    except IndexError:
        raise click.UsageError("Invalid verbosity level (use -v, -vv, or -vvv)")
```

**Comment by originalsouth** [comment]
> @originalsouth Could you update the PR title? 😁

Why? What is wrong with "Rewrite xtdb-cli.py with "click"" which is exactly what this patch achieves.

**Review by underdarknl — APPROVED** [review]

**Pull request merged**